### DISCUSSION TITLE
How to install using Docker?

### DISCUSSION THREAD

**Discussion by AndreaPi** [hlky/stable-diffusion-webui#778] [Q&A] [answered]
Is there a guide to install `stable-diffusion-webui` using Docker? I cloned the repository on a server where Docker is installed. What should I do now? I see the following Docker-related files:

```
.env_docker.example
docker-compose.yml
docker-reset.sh
Dockerfile
entrypoint.sh
```

I edited the `Dockerfile` a bit so that it would work in my env (mostly passing the necessary certificates). `docker-compose.yml` seems to include a build stage, so I guess I should just run

`docker-compose up -d`

and Docker should build & run the container. However, in the yml file there's a reference to a non-existing `.env_docker` file. Is it enough to rename `.env_docker.example` to `.env_docker`? Are there any particular options I should change? In particular, I see this line:
```
WEBUI_ARGS=
```

is it fine to leave this args list empty? If not, which are some typical arguments I should set? I'm running on a system with a NVIDIA GPU.

**Comment by john-warnes** [comment, marked as answer]
I didn't keep notes but got mine working with something like.... 

On linux I had install the NVIDIA Container Toolkit for it to run and access the GPU, I have no idea about windows, check here: `https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/install-guide.html`
Might be builtin

Copy `.env_docker.example` -> `.env_docker` 
edit file as needed: `WEBUI_ARGS=--save-metadata`

run `docker-compose up -d` or `docker compose up -d`

**Reply by AndreaPi** [reply]
Nice! I will test this and let you know. Thanks!

**Reply by AndreaPi** [reply]
Seems to work, but I'm having an error during build:

```
Step 9/16 : RUN apt-get update &&     apt-get install -y libglib2.0-0 wget &&     apt-get clean &&     rm -rf /var/lib/apt/lists/*
 ---> Running in 377a88c04cf2
Get:1 http://security.ubuntu.com/ubuntu focal-security InRelease [114 kB]
Get:2 http://archive.ubuntu.com/ubuntu focal InRelease [265 kB]
Ign:3 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64  InRelease
Get:4 http://security.ubuntu.com/ubuntu focal-security/main amd64 Packages [2129 kB]
Get:5 http://archive.ubuntu.com/ubuntu focal-updates InRelease [114 kB]
Get:6 http://archive.ubuntu.com/ubuntu focal-backports InRelease [108 kB]
Err:7 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64  Release
  Certificate verification failed: The certificate is NOT trusted. The certificate issuer is unknown.  Could not handshake: Error in the certificate verification. [IP: 152.195.19.142 443]
Get:8 http://security.ubuntu.com/ubuntu focal-security/universe amd64 Packages [892 kB]
Get:9 http://archive.ubuntu.com/ubuntu focal/universe amd64 Packages [11.3 MB]
Get:10 http://security.ubuntu.com/ubuntu focal-security/restricted amd64 Packages [1501 kB]
Get:11 http://security.ubuntu.com/ubuntu focal-security/multiverse amd64 Packages [27.5 kB]
Get:12 http://archive.ubuntu.com/ubuntu focal/main amd64 Packages [1275 kB]
Get:13 http://archive.ubuntu.com/ubuntu focal/multiverse amd64 Packages [177 kB]
Get:14 http://archive.ubuntu.com/ubuntu focal/restricted amd64 Packages [33.4 kB]
Get:15 http://archive.ubuntu.com/ubuntu focal-updates/multiverse amd64 Packages [30.2 kB]
Get:16 http://archive.ubuntu.com/ubuntu focal-updates/universe amd64 Packages [1194 kB]
Get:17 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 Packages [2590 kB]
Get:18 http://archive.ubuntu.com/ubuntu focal-updates/restricted amd64 Packages [1613 kB]
Get:19 http://archive.ubuntu.com/ubuntu focal-backports/universe amd64 Packages [27.4 kB]
Get:20 http://archive.ubuntu.com/ubuntu focal-backports/main amd64 Packages [55.1 kB]
Reading package lists...
E: The repository 'https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64  Release' does not have a Release file.
```

 I'll open a issue and close this discussion for now.

**Reply by oc013** [reply]
I recreated my docker guide here #922
